src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindIntegrationTest.java

```diff
@@ -1269,6 +1270,143 @@ public void byDateColumn() {
+                        }
+                    }
+                }
+
+              """;
+
+      String expected1 =
+          """
+                  {"_id":"doc1", "username":"user1", "active_user":true, "date" : {"$date": 1672531200000}}
+                  """;
+      String expected2 =
+          """
+                  {"_id":"doc2", "username":"user2", "subdoc":{"id":"abc"},"array":["value1"]}
+                  """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(2));
+    }
+
+    @Test
+    public void nestedAndOr() {
+      String json =
+          """
+            {
+                "find": {
+                    "filter": {
+                        "$and": [
+                            {
+                                "$or": [
+                                    {
+                                        "username": "user3"
+                                    },
+                                    {
+                                        "subdoc.id": {
+                                            "$eq": "abc"
+                                        }
+                                    }
+                                ]
+                            },
+                            {
+                                "$or": [
+                                    {
+                                        "username": "user2"
+                                    },
+                                    {
+                                        "subdoc.id": {
+                                            "$eq": "xyz"
+                                        }
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                }
+            }
+
+              """;
+
+      String expected1 =
+          """
+                  {"_id":"doc1", "username":"user1", "active_user":true, "date" : {"$date": 1672531200000}}
+                  """;
+      String expected2 =
+          """
+                  {"_id":"doc2", "username":"user2", "subdoc":{"id":"abc"},"array":["value1"]}
+                  """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(1));
```

Comment:
Can we test that the document is equal to the expected result?

Reply:
solved, thanks